docs/config.rst

```diff
@@ -189,23 +189,50 @@ by using the ``tox:jenkins`` section:
     commands = ...  # override settings for the jenkins context
 
 
-tox environment settings
-------------------------
+tox environments
+----------------
 
-Test environments are defined by a:
+Test environments are defined under the ``testenv`` section and individual
+``testenv:NAME`` sections, where ``NAME`` is the name of a specific
+environment.
 
 .. code-block:: ini
 
+    [testenv]
+    commands = ...
+
     [testenv:NAME]
     commands = ...
 
-section.  The ``NAME`` will be the name of the virtual environment.
-Defaults for each setting in this section are looked up in the::
+Settings defined in the top-level ``testenv`` section are automatically
+inherited by individual environments unless overridden. Test environment names
+can consist of alphanumeric characters and dashes; for example:
+``py38-django30``. The name will be split on dashes into multiple factors,
+meaning ``py38-django30`` will be split into two factors: ``py38`` and
+``django30``. *tox* defines a number of default factors, which correspond to
+various versions and implementations of Python and provide default values for
+:conf:`basepython`:
+
+- ``pyNM``: configures ``basepython = pythonN.M``
+- ``pyN``: configures ``basepython = pythonN``
+- ``py``: configures ``basepython = python``
+- ``pypyN``: configures ``basepython = pypyN``
+- ``pypy``: configures ``basepython = pypy``
+- ``jythonN``: configures ``basepython = jythonN``
+- ``jython``: configures ``basepython = jython``
+
+It is also possible to define what's know as *generative names*, where an
+individual section maps to multiple environments; for example:
+``py{37,38}-django{30,31}``, which would generate four environments, each
+consisting of two factors a piece: ``py37-django30`` (``py37``, ``django30``),
+``py37-django31`` (``py37``, ``django31``), ``py38-django30`` (``py38``,
+``django30``), and ``py38-django31`` (``py38``, ``django31``).  Combined, these
+features provide the ability to write very concise ``tox.ini`` files and is
+discussed further `below <generating-environments>`__.
```

Comment:
This "below" link doesn't work:

From https://tox.readthedocs.io/en/latest/config.html#tox-environments section, it leads to https://tox.readthedocs.io/en/latest/generating-environments "SORRY This page does not exist yet."

Reply:
Damn, this probably should have used `:ref:` rather than the double underscore